k8s/k8s.go

```diff
@@ -131,21 +150,43 @@ func CmdAddK8s(args *skel.CmdArgs, conf utils.NetConf, hostname string, calicoCl
 			logger.WithField("labels", labels).Info("Fetched K8s labels")
 			endpoint.Metadata.Labels = labels
 		}
+	}
 
-		fmt.Fprintf(os.Stderr, "Calico CNI using IPs: %s\n", endpoint.Spec.IPNetworks)
+	// Whether the endpoint existed or not, ensure that its IP Networks
+	// are populated with the correct information.
+	if err = utils.PopulateEndpointNets(endpoint, result); err != nil {
+		// Cleanup IP allocation and return the error.
+		utils.ReleaseIPAllocation(logger, conf.IPAM.Type, args.StdinData)
+		return nil, err
 	}
+	logger.WithField("endpoint", endpoint).Info("Populated endpoint")
+	fmt.Fprintf(os.Stderr, "Calico CNI using IPs: %s\n", endpoint.Spec.IPNetworks)
 
 	// Whether the endpoint existed or not, the veth needs (re)creating.
-	hostVethName, contVethMac, err := utils.DoNetworking(args, conf, result, logger)
+	// Pass the existing veth name, if it is present, so it can be created
+	// with the same name.
+	var prevVethName *string
```

Comment:
I think I want to get rid of this behavior, and instead call into a common function `vethParamsForWorkload` which returns the veth name and mac.

This will help remove all the datastore-specific logic I've had to add to get this working.